Crash starting on gl driver

Dolphin is crashing on gl (glcore is fine) with the driver switching code.

```
[ERROR] GL: Invalid enum.
[ERROR] [Video]: Cannot open video driver ... Exiting ...
[ERROR] Fatal error received in: "init_video()"
[ERROR] failed_to_start_audio_driver
```
[retroarch.log](https://github.com/libretro/dolphin/files/3775119/retroarch.log)

It start when disabling the setting in Core>allow cores to switch...

win7 x64 nvidia GTX 770


@Rinnegatamante

The Dolphin emulator is no longer opening GameCube games with gl driver.

```
[ERROR] GL: Invalid enum.
[ERROR] [Video]: Cannot open video driver ... Exiting ...
[ERROR] Fatal error received in: "init_video()"
```

:´(

--- a/Source/Core/DolphinLibretro/Video.cpp
+++ b/Source/Core/DolphinLibretro/Video.cpp
@@ -230,7 +230,7 @@ void Init()
       if (preferred != 0xFFFFFFFF) {
          hw_render.version_major = 3;
          hw_render.version_minor = 1;
-         hw_render.context_type = (retro_hw_context_type)preferred;
+         hw_render.context_type = RETRO_HW_CONTEXT_OPENGL_CORE;
          if (environ_cb(RETRO_ENVIRONMENT_SET_HW_RENDER, &hw_render))
          {
            environ_cb(RETRO_ENVIRONMENT_SET_HW_SHARED_CONTEXT, nullptr);

